File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeNodes(ListNode* head) {
        ListNode *temp, *cur = head;
        ListNode* dummy = new ListNode(0);
        temp = dummy;
        while(cur->next) {
            int sum = 0;
            while(cur->next->val != 0) {
                sum += cur->next->val;
                cur = cur->next;
            }
            cur = cur->next;
            temp->next = new ListNode(sum);
            temp = temp->next;
        }
        return dummy->next;
    }
};

File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeNodes(ListNode head) {
        ListNode cur = head;
        ListNode dummy = new ListNode(0);
        ListNode temp = dummy;
        while(cur.next != null) {
            int sum = 0;
            while(cur.next.val != 0) {
                sum += cur.next.val;
                cur = cur.next;
            }
            temp.next = new ListNode(sum);
            temp = temp.next;
            cur = cur.next;
        }
        return dummy.next;
    }
}